swarm/storage/pyramid.go

```diff
@@ -75,15 +75,61 @@ const (
 	TreeChunk = 1
 )
 
-type ChunkerParams struct {
-	Branches int64
-	Hash     string
+type PyramidSplitterParams struct {
+	SplitterParams
+	getter Getter
 }
 
-func NewChunkerParams() *ChunkerParams {
-	return &ChunkerParams{
-		Branches: DefaultBranches,
-		Hash:     SHA3Hash,
+func NewPyramidSplitterParams(key Key, reader io.Reader, putter Putter, getter Getter, branches int64) *PyramidSplitterParams {
+	hashSize := putter.RefSize()
+	return &PyramidSplitterParams{
+		SplitterParams: SplitterParams{
+			ChunkerParams: ChunkerParams{
+				// TODO: use chunkSize as parameter, because that is the constant
+				branches: 4096 / hashSize,
```

Comment:
Maybe extract 4096 as a constant and document it, since it is used at at least 3 places.

Reply:
I agree, that is why I have the TODO in the line before, I'm fixing this.

Reply:
@nonsense Fixed in 68cf01eb4fe28e729a1b32eafe1959c28a1b36a4